src/common/desktopenvcmn.cpp

```diff
@@ -0,0 +1,40 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/common/desktopenv.cpp
+// Purpose:     implementation of wxDesktopEnvBase
+// Author:      Igor Korot
+// Created:     2021-09-05
+// Copyright:   (c) 2021 wxWidgets development team
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+// ============================================================================
+// declarations
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+// for compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#include "wx/filefn.h"
+#include <wx/filename.h>
+#include "wx/log.h"
+#include "wx/desktopenv.h"
+
+/* static */
+bool wxDesktopEnvBase::MoveToRecycleBin(const wxString &path)
+{
+    bool result = true;
+    wxFileName name = wxFileName( path );
+
+    name.MakeAbsolute();
+    wxString tempPath = name.GetFullPath();
+    if( !wxFileExists( tempPath ) && !wxDirExists( tempPath ) )
+    {
+        wxLogSysError( _( "Failed to move '%s' to Recycle Bin" ), path );
+        result = false;
+    }
```

Comment:
Why should we bother checking this at all? If the file doesn't exist, moving it to trash should fail anyhow.

This entire function should be just removed.

Reply:
Vadim,
Why even try if the file doesn't exist? It just doesn't make sense...
And this is exactly why the base class exist - to have this check in one place and don't have it in 3 places.

I may give it a try and see if it succeeds on all 3 supported platforms without crashing the program.